stateMutators.js

```diff
@@ -0,0 +1,109 @@
+const gameState = {
+    basketPos: 50,
+    basketValue: 0,
+    score: 0,
+    livesRemaining: 3,
+    gameLevel: 1,
+    gameMode: 'running',
+    fallingObjects: [],
+}
+
+const setBasketPos = (state, value) => ({
+    ...state,
+    basketPos: value,
+});
+
+const moveBasketLeft = (state, value) => ({
+    ...state,
+    basketPos: basketPos - value,
+});
+
+const moveBasketRight = (state, value) => ({
+    ...state,
+    basketPos: basketPos + value,
+});
+
+const moveBasketMouse = (state, value) => ({
+    ...state,
+    basketPos: value,
+});
+
+const setBasketValue = (state, value) => ({
+    ...state,
+    basketValue: value,
+});
+
+const resetBasketValue = (state) => ({
+    ...state,
+    basketValue: 0,
+});
+
+const setScore = (state, value) => ({
+    ...state,
+    score: value,
+});
+
+const resetScore = (state) => ({
+    ...state,
+    score: 0,
+});
+
+const loseLife = (state, value) => ({
+    ...state,
+    livesRemaining: value,
+});
+
+const resetLivesRemaining = (state) => ({
+    ...state,
+    livesRemaining: 3,
+});
+
+const setGameLevel = (state, value) => ({
+    ...state,
+    gameLevel: value,
+});
+
+const resetGameLevel = (state) => ({
+    ...state,
+    gameLevel: 0,
+});
+
+const setGameMode = (state, value) => ({
+    ...state,
+    gameMode: value,
+});
+
+const addFallingObject = (state, fallingObj) => ({
+    ...state,
+    fallingObjects: state.fallingObjects.push(fallingObj)
```

Comment:
`Array.push` here actually returns a `number` value denoting the number of elements in the array rather than the array itself. Which means this mutation is going to overwrite the `fallingObjects` state with a single number value when this gets called. Docs: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/push#return_value

I can think of two ways you could update it to do what you want here:

1. using array destructuring to effectively add the new obj to the add of the array:
```js
const addFallingObject = (state, fallingObj) => ({
    ...state,
    fallingObjects: [
		...state.fallingObjects,
		fallingObj,
    ],
})
```

2. using `Array.push` to add it in a separate line, then return the changed state in the next line
```js
const addFallingObject = (state, fallingObj) => {
	state.fallingObjects.push(fallingObj)
	return state
}
```

I personally would do 1 as it creates a brand new array and state object each call (if your mutations return brand new objects every time they're called, you can do some fancy stuff like travel back and forwards in time through your state mutations, which can be useful for debugging and other things - though a bit more advanced concepts), and I find the `...` destructuring notation more self-descriptive of the data changes (though possibly subjective).